go/oasis-node/cmd/genesis/genesis.go

```diff
@@ -41,6 +47,34 @@ const (
 	cfgBlockHeight        = "height"
 	cfgChainID            = "chain.id"
 
+	// Registry config flags.
+	cfgRegistryDebugAllowUnroutableAddresses = "registry.debug.allow_unroutable_addresses"
+	cfgRegistryDebugAllowRuntimeRegistration = "registry.debug.allow_runtime_registration"
+	cfgRegistryDebugBypassStake              = "registry.debug.bypass_stake" // nolint: gosec
+
+	// Roothash config flags.
+	cfgRoundTimeout = "roothash.round_timeout"
+
+	// Scheduler config flags.
+	cfgSchedulerDebugBypassStake      = "scheduler.debug.bypass_stake" // nolint: gosec
+	cfgSchedulerDebugStaticValidators = "scheduler.debug.static_validators"
+	cfgSchedulerAlgorithm             = "worker.txnscheduler.algorithm"
+	cfgSchedulerBatchFlushTimeout     = "worker.txnscheduler.flush_timeout"
+
+	// Beacon config flags.
+	cfgBeaconDebugDeterministic = "beacon.debug.deterministic"
+
+	// EpochTime config flags.
+	cfgEpochTimeBackend            = "epochtime.backend"
```

Comment:
possibly also `consensus.backend` (not that we support any other than tendermint atm :P)

Reply:
This will probably need some refactoring to make it work -- we read the genesis document as part of the tendermint initialization, so we'd have to move that to an earlier point.  I'd rather leave this for a separate PR/issue :)

Reply:
We do have our own genesis provider, it is not tied to Tendermint. See `go/genesis`. I think that one is initialized in the oasis-node binary so you could in theory look at it before initializing Tendermint.